pkg/netceptor/packetconn.go

```diff
@@ -25,9 +41,30 @@ type PacketConn struct {
 	cancel          context.CancelFunc
 }
 
+func NewPacketConnWithConst(s *Netceptor, service string, advertise bool, adtags map[string]string, connTypeDatagram byte) (*PacketConn, error) {
+	npc := &PacketConn{
+		s:            s,
+		localService: service,
+		recvChan:     make(chan *MessageData),
+		advertise:    advertise,
+		adTags:       adtags,
+		connType:     connTypeDatagram,
+		hopsToLive:   s.maxForwardingHops,
+	}
+
+	npc.startUnreachable()
+	s.listenerRegistry[service] = npc
+
+	return npc, nil
+}
+
+func NewPacketConn(s *Netceptor, service string, connTypeDatagram byte) (*PacketConn, error) {
```

Comment:
Same comment for `Netceptor` as above

Reply:
We will create another PR to add the Netceptor interface as it will increase the complexity and focus of this PR.